### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by martinkarugaba**
# Changes made

## Added functionality to handle

- [ ] Create post
- [ ] Add comment to post
- [ ] Add likes to post

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <%= f.text_field :title %>
+    </div>
+    <div class='mt-8'>
+      <p class='font-medium'><%= f.label :text %><p>
+      <%= f.text_area :text %>
+    </div>  
+  </div>
+
+  <div>
+    <%= f.submit 'Create Post' class='mt-8 p-2 cursor-pointer boder-2 border-black rounded-md' %>
```
Comment (Strangeal) [inline app/views/posts/new.html.erb]: - [ ] 👍 Great progress on building the forms for your blog app! 🚀 However, I noticed a syntax error in your post form related to the styling of the submit button. To enhance readability and resolve this issue, consider applying the styles directly to a `button` element and encapsulating it around the `f.submit` button. This ensures a cleaner structure and avoids potential syntax conflicts.

**Here's a revised version for your consideration:**

```ruby
<%= button_tag(type: 'submit', class: 'mt-8 p-2 cursor-pointer boder-2 border-black rounded-md') do %>
  Create Post
<% end %>
```

This modification separates the styling concerns, making your code more maintainable and syntactically correct.  Happy coding!😊👨‍💻🚀

File: app/views/_post.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<div class="post border p-4 rounded-md border-black">
+  <h3><%= post.title %></h3>
+  <p><%= post.text %></p>
+  <div class='my-4 flex justify-end items-center gap-4 text-sm'>
+    <p>Comments: <%= post.comments.count %></p>
+    <div class='my-4 flex justify-end items-center gap-4 text-sm'>
+      <p>
+        <%= button_to 'Like', like_user_post_path(@user, post), remote: true, method: :post, class: 'bg-blue-500 px-4 py-2 cursor-pointer rounded-md text-white', data: { post_id: post.id } %><span id='likes-count'>
```
Comment (Strangeal) [inline app/views/_post.html.erb]: - [ ] Fantastic progress on your implementation! 🚀 Your hard work is evident, but let's ensure we cover a crucial requirement: the `like` functionality. 🤝 After reviewing the project requirements, it seems we might have missed incorporating this feature.

**Here's a screenshot of the requirement for your reference:** 
Screenshot 2023-12-15 at 21 28 13


 - To enhance the user experience, it's important to implement the `like` functionality as outlined. This feature not only aligns with the project specifications but also adds a valuable element for user engagement.

Happy coding! 😊👨‍💻🚀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @martinkarugaba, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Pixel Perfect Design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/_post.html.erb (hunk 2 above)
Comment (martinkarugaba) [reply app/views/_post.html.erb]: Hi @Strangeal my like functionality is working, did you try clicking the like button

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::Base
+  helper_method :current_user
+
+  private
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
```
Comment (LionRouge1) [inline app/controllers/application_controller.rb:8]: [OPTIONAL]
- [ ] Nice work implementing the `current_user` method in the `applicationController` class :clap: However, according to the project requirements the `current_user` method should return the first user from the database. Therefore I suggest that you update the the method to return the first user from the database to follow the project requirement.

Example:
```rb
...

def current_user
  User.first
end
```
_This is a requirement of the project but the first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change_

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -3,11 +3,43 @@ class PostsController < ApplicationController
...
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.create(user: current_user, post: @post)
+
+    respond_to do |format|
+      format.json { render json: { likes_count: @post.likes.count } }
+    end
+  end
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb:27]: [OPTIONAL]
- [ ] Kindly remove all actions related to the **Likes** into the `app/controllers/likes_controller.rb` file to make your project look more clean.

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -3,11 +3,43 @@ class PostsController < ApplicationController
...
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build(post_params)
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb:31]: [OPTIONAL]
- [ ] According to the project requirement we should created a post on behalf of the currently logged in user. Therefore I suggest that you make use of the `current_user` method that you created in the `applicationController` to set the user for a new post.
  
Example:
```rb
@post = current_user.posts.build(post_params)
```

_This is a requirement of the project but first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change_

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    @post = @user.posts.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.user = @user
+    @comment.post = @post
```
Comment (LionRouge1) [inline app/controllers/comments_controller.rb:13]: - [ ] Same thing here as I pointed in the previous comment, Kindly make use of the `current_user` method created in the `ApplicationController` to set the user for a comment

  Example:
```rb
    @post = @user.posts.find(params[:post_id])
    @comment = current_user.comments.new(comment_params)
    @comment.post = @post
```

_This is a requirement of the project but first reviewer didn't notice it. so I requested this as an optional but I highly recommend you to make the change_

**Review by LionRouge1 — APPROVED** [review]
Hi @martinkarugaba, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Highlights
- Great job, your project is error free :+1: 
- Nice work you meet all the requirements :1st_place_medal: 
- All linters passed :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 6 above)
Comment (martinkarugaba) [reply app/controllers/comments_controller.rb:13]: Thank you, @LionRouge1 I'll look into all those items

**Pull request merged**